### DISCUSSION TITLE
A way to replace the topbar with custom one

### DISCUSSION THREAD

**Discussion by gavinhewitt** [filamentphp/filament#13927] [Ideas]
### Package

Panel builder

### Package Version

3.*

### How can we help you?

We currently only have render hooks but it would be great also to be able to replace the whole topbar with our own custom one.

**Comment by rvzug** [comment]
You can publish the blade views, but read this disclaimer first: https://filamentphp.com/docs/3.x/support/style-customization#publishing-blade-views

**Reply by gavinhewitt** [reply]
Yes. Probably should have chosen the 'idea' label because it's a suggestion and not a question. Thanks anyway 😉

**Reply by gavinhewitt** [reply]
Changed to idea

**Comment by helloreece** [comment]
->topbar(false) will get rid of the existing top bar.

Using a render hook is then a pretty simple way of implementing your own navigation bar.

**Reply by gavinhewitt** [reply]
Yes, removing the topbar and then hooking your custom component could be a way to do it. However, if you look at `vendor/filament/filament/resources/views/components/layout/index.blade.php` there is no hook in that exact place for a custom topbar to be rendered, so that doesn't fly.
